tests/test_g2g_communication.py

```diff
@@ -85,7 +85,7 @@ def gpu_communicators(cube_partitioner):
 
 
 @contextlib.contextmanager
-def module_count_calls_to_empty(module):
+def module_count_calls_to_ones(module):
     global N_EMPTY_CALLS
     N_EMPTY_CALLS[module.empty] = 0
```

Comment:
This code needs to be updated, too.

Reply:
No why? The functional API uses the new object under the hood so that code path still tests the G2G and force CPU behavior

Reply:
If the module is supposed to count calls to `np.ones`, why is it setting the count for `module.empty` to zero? Shouldn't it be `module.ones`, and shouldn't the dict be called `N_ONES_CALLS`?

Reply:
Yes, shortly after writing this I found a bug on that very feature -_-. Fixed the bug & made the utest use zeros while the internal call to zero that isn't buffer is switch to empty